From 8f5fa261d83457a3406ea4962e2505596f7159e6 Mon Sep 17 00:00:00 2001
From: Kyrylo Mushta <k.mushta@inango-systems.com>
Date: Tue, 8 Jun 2021 21:38:44 +0300
Subject: [PATCH] Commit Title          : Update init configuration for enable
 offload features in ovs

BugID                 : 24388
Bug Fixed(y/N[blankN]):
CodeReviewId          : 0
Description           : Updated init configuration for enable offload features in ovs
* enabled proactive mode during init ovs-vswitchd
* set tcp flags always wildcards
* disabled megaflow
* disabled hw-offload
---
 utilities/ovs-ctl.in | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/utilities/ovs-ctl.in b/utilities/ovs-ctl.in
index ee27cc62c..fedc5947b 100644
--- a/utilities/ovs-ctl.in
+++ b/utilities/ovs-ctl.in
@@ -247,8 +247,17 @@ do_start_brcompatd () {
         start_daemon "$OVS_BRCOMPATD_PRIORITY" "$OVS_BRCOMPATD_WRAPPER" "$@" ||
             return 1
 
-        action "Enabling the flow hardware offload" \
-            ovs-vsctl set Open_vSwitch . other_config:hw-offload=true
+        action "Enabling proactive mode" \
+            ovs-appctl upcall/enable-proactive-mode
+
+        action "Set tcp flags always wildcards" \
+            ovs-appctl upcall/enable-megaflows-mask-fields tcp_flags
+
+        action "Disabling megaflows in OVS datapath" \
+            ovs-appctl upcall/disable-megaflows
+
+        action "Disabling the flow hardware offload" \
+            ovs-vsctl set Open_vSwitch . other_config:hw-offload=false
     fi
 }
 
